lib/dealership.rb

```diff
@@ -0,0 +1,44 @@
+class Dealership
+  attr_reader :name, :address, :inventory, :inventory_count
+  def initialize(name, address)
+    @name = name
+    @address = address
+    @inventory = []
+    @inventory_count = 0
+  end
+
+  def add_car(car)
+    @inventory << car
+    @inventory_count += 1
+  end
+
+  def has_inventory?
+    if @inventory_count == 0
+      false
+    else
+      true
+    end
+  end
+
+  def cars_by_make(make)
+    @inventory.select do |car|
+      car.make == make
+    end
+  end
+
+  def total_value
+    @total = []
+    @inventory.each do |car|
+      @total << car.total_cost
+    end
+    @total.sum
+  end
+
+  def details
+    # require 'pry'; binding.pry
```

Comment:
Super close here. So since we are never needing to access this hash, you don't have to set it to a variable. You also do not need to call `@dealership.total_value` and `@dealership.address`. You can just call put those methods in.
```
{
    "total_value" => total_value,
    "address" => @address
}
```